### ISSUE TITLE
STPPaymentOptionsViewControllerDelegate didSelect returns a default paymentOption

### ISSUE THREAD

**Issue by gilrodar** [stripe/stripe-ios#1638]
## Summary

When using the "didSelect paymentOption" method of the STPPaymentOptionsViewControllerDelegate it returns a paymentOption since the STPPaymentOptionsViewController is called and does not let you select a new payment method.

## Code to reproduce

```
func paymentOptionsViewController(_ paymentOptionsViewController: STPPaymentOptionsViewController, didSelect paymentOption: STPPaymentOption) {
        print(paymentOption)
        dismiss(animated: true)
}
```

## iOS version

iOS 13.7

## Installation method

Cocoapods

## SDK version

19.4.0

**Comment by aliriaz-stripe** [comment]
Hello @gilrodar , could you explain more? What issue are you running into? 

In particular, I didn't fully understand what you mean by 
> since the STPPaymentOptionsViewController is called and does not let you select a new payment method.

**Comment by gilrodar** [comment]
I present the STPPaymentOptionsViewController to select a new payment method for a user, but it does not allow me to make the selection since it returns a "paymentOption" when the controller is displayed.

**Comment by aliriaz-stripe** [comment]
@gilrodar would it be possible to share code for how you're setting up your STPPaymentOptionsViewController? 

And log out when you hit your delegate method? 

I can take a look

**Comment by aliriaz-stripe** [comment]
@gilrodar hello! Just a bump, are you still blocked on this? 

Happy to see some code if you want to share via a gist or pastebin, so I can see what you're setting up.
Also helpful if you describe what behavior you're seeing, versus what you would like to see?

**Comment by vincentcastagna** [comment]
@aliriaz-stripe We have a similar issue on a Xamarin Binding. It seems like the kLastSelectedPaymentMethodDefaultsKey , which stores the latest selected payment method in NSUserDefaults causes the behavior.
Cause as long as the user has selected a payment method before, re-triggering the PaymentOptionsViewController  would simply open it and close again with the previous selectedpayment, without giving the user the possibility to add or select another card. 

We workarounded this by removing the key from NSUserDefaults and then synchronize but it seems that it introduces some other inconsistent behavior for us when adding/selecting another card.
